### PR TITLE
Added OSC Layout and Visibility Configurability, #495, #450, #392, and #389

### PR DESCRIPTION
**Pull request by King0fSpace**
This commit implements a feature discussed in #495 , #450 , #392 , and #389 . 

It allows users to hide and move any of the on screen controller buttons anywhere they’d like on screen. The user can save their custom layouts under user-defined names. Multiple controller ‘profiles’ can be saved. Profiles can be loaded from a menu that lists all of the user’s saved profiles. The user can swipe-to-delete any user-saved profiles they'd like from this screen as well.

This feature has been made accessible in an unobtrusive way; by tapping the ‘Custom’ button added to the end of the existing ‘On-Screen Controls’ buttons bar, located within the existing ‘Settings’ slide out menu. 

IMG_4526
IMG_4465
IMG_4466
7c93.jpeg)
IMG_4464
IMG_4463
IMG_4490
IMG_4470
IMG_4472

iPad
IMG_0047

### CODE REVIEW

**Comment by cgutman** [comment]
Sorry for the delays reviewing this. I've been trying to get fixes out for various Moonlight clients.

I will try to take a look by this weekend.

**Comment by King0fSpace** [comment]
No worries. I love the app. Thanks very much

File: .gitmodules.save [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "moonlight-common/moonlight-common-c"]
+	path = moonlight-common/moonlight-common-c
+	url = https://github.com/moonlight-stream/moonlight-common-c
```
Comment (cgutman) [inline .gitmodules.save:3]: Looks like this file was committed accidentally

File: .gitmodules [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [submodule "moonlight-common/moonlight-common-c"]
 	path = moonlight-common/moonlight-common-c
-	url = https://github.com/moonlight-stream/moonlight-common-c.git
+	url = https://github.com/moonlight-stream/moonlight-common-c
 [submodule "X1Kit"]
 	path = X1Kit
-	url = https://github.com/cgutman/X1Kit.git
+	url = https://github.com/cgutman/X1Kit.git
```
Comment (cgutman) [inline .gitmodules:6]: Can you revert these? They don't seem relevant to the PR here.

File: .gitignore [hunk 3]
```diff
@@ -1,5 +0,0 @@
-**/xcuserdata/
-**/Moonlight.xcscmblueprint
-**/xcschemes/*.xcscheme
-Build
-DerivedData
```
Comment (cgutman) [inline .gitignore:5]: Please revert this removal

File: .DS_Store
Comment (cgutman) [inline .DS_Store:1]: Please remove these `.DS_Store` files

**Review by cgutman — CHANGES_REQUESTED** [review]
It looks like there are a bunch of extraneous files moved/deleted/added that aren't related to the change here (ex: `.DS_Store`, all of `moonlight-common-c`, `pbxproj` files, etc).

Can you rebase your branch to only include the relevant changes for your feature?

**Comment by King0fSpace** [comment]
Opened another pull request

**Pull request closed without merging**